tests/utils/test_rest_util.py

```diff
@@ -454,6 +454,34 @@ def test_proxy_conditions_with_no_proxy(self):
             host = "10.0.1.1"
             self.assertEquals(should_use_proxy, use_proxy and not restutil.bypass_proxy(host))
 
+    def test_proxy_conditions_with_no_proxy_not_set(self):
+        should_use_proxy = True
+        use_proxy = True
+
+        host = "10.0.0.1"
+        self.assertEquals(should_use_proxy, use_proxy and not restutil.bypass_proxy(host))
+
+        host = "foo.com"
+        self.assertEquals(should_use_proxy, use_proxy and not restutil.bypass_proxy(host))
+
+        host = "www.google.com"
+        self.assertEquals(should_use_proxy, use_proxy and not restutil.bypass_proxy(host))
+
+        host = "168.63.129.16"
+        self.assertEquals(should_use_proxy, use_proxy and not restutil.bypass_proxy(host))
+
+        host = "www.bar.com"
+        self.assertEquals(should_use_proxy, use_proxy and not restutil.bypass_proxy(host))
+
+        host = "www.bar.com"
+        self.assertEquals(should_use_proxy, use_proxy and not restutil.bypass_proxy(host))
+
+        host = "10.0.0.1"
+        self.assertEquals(should_use_proxy, use_proxy and not restutil.bypass_proxy(host))
+
+        host = "10.0.1.1"
+        self.assertEquals(should_use_proxy, use_proxy and not restutil.bypass_proxy(host))
```

Comment:
I know this is just a copy of the test above, but the assert is kinda hard to read.. why not simply AssertTrue (or AssertFalse)

Reply:
I just wanted to make it explicit that proxy should be used in these cases (when no_proxy is not set). I added a comment and refactored to make it cleaner.